ui/hooks/useNftCollectionsMetadata.ts

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useMemo, useState } from 'react';
+import { Collection } from '@metamask/assets-controllers';
+import { useSelector } from 'react-redux';
+import { TokenStandard } from '../../shared/constants/transaction';
+import { getCurrentChainId } from '../selectors';
+import {
+  getNFTContractInfo,
+  getTokenStandardAndDetails,
+} from '../store/actions';
+
+export type UseNftCollectionsMetadataRequest = {
+  value: string;
+};
+
+type CollectionsData = {
+  [key: string]: Collection;
+};
+
+// For now, we only support ERC721 tokens
+const SUPPORTED_NFT_TOKEN_STANDARDS = [TokenStandard.ERC721];
+
+export function useNftCollectionsMetadata(
+  requests: UseNftCollectionsMetadataRequest[],
+) {
+  const [collectionsMetadata, setCollectionsMetadata] =
+    useState<CollectionsData>({});
+  const chainId = useSelector(getCurrentChainId);
+
+  const memoisedContracts = useMemo(() => {
+    return requests
+      .filter(({ value }) => value)
+      .map(({ value }) => value.toLowerCase());
+  }, [requests]);
+
+  useEffect(() => {
+    const fetchCollections = async () => {
```

Comment:
Minor, but since this is a large function, we could define it outside of the component and `useEffect` (but still in this file) for greater readability, and even split it up further if needed.

Then we could also use `useAsyncResult` to handle cancellation on subsequent renders, and also detect loading state.

Reply:
Fixed this.